docs/development/performance/notifications.rst

```diff
@@ -0,0 +1,66 @@
+.. _performance_notifications:
+
+Notifications performance
+=========================
+
+Open Zaak sends notifications using `Celery`_ as a task queue.
+Since sending notifications is asynchronous we use auto retry in case the tasks have failed.
+
+Below we analyze the reliability and performance of sending notifications about
+the creation of the zaak.
+
+
+Sending notifications under burst load
+--------------------------------------
+
+We measure the notifications median time and failure rate under the increasing
+amount of concurrent users.
+
+**Test specifications:**
+
+* 1 celery worker
+* no waiting time between requests for 1 user
+* all users send requests to create zaak (``POST /api/v1/zaken``)
+* time under load is 1 minute
+
+.. csv-table:: Notification performance results per users
+   :header-rows: 1
+
+    Users,Request Count,Requests/s,Failure Count,Median Response Time (ms),Notifications failure,Median Notification Time (ms)
+    10,1430,24.183,0,390,0,88.572
+    20,1479,25.034,0,740,0,83.922
+    40,1429,24.189,0,1200,0,86.557
+
+The results of the burst load test are promising. We have 0 failure rate both for zaak
+creation and for sending of notifications about it.
+
+.. _Celery: https://docs.celeryq.dev/en/stable/
+
+
+Automatic retry for notifications
+---------------------------------
+
+Sending notifications can be done with automatic retry, i.e. if the notification task was failed it
+will be autoretried the configured number of times. We measure how autoretry works with increasing downtime
+of the notification service.
+
+**Test specifications:**
+
+* 1 celery worker
+* 10 concurrent users
+* no waiting time between requests for 1 user
+* all users send requests to create zaak (``POST /api/v1/zaken``)
+* requests are send during 1 min
+* auto retry settings are default (5 maximum attempts per task)
+
+.. csv-table:: Notification autoretries per downtime
+   :header-rows: 1
+
+    Notification downtime (s),Request Count,Failure Count,Notifications failed,Notifications retried,Notifications processed
+    10,1486,0,0,919,2114
+    60,1478,0,0,3729,5195
+    300,1509,0,1276,6380,7908
+
+According to the result if the notification server is down for less than 1 min, then all notifications
+would be eventually send. If it's expected that the notification service is down for longer time
+the autoretry configuration should be adjusted in the admin.
```

Comment:
I'm having a bit of a hard time interpreting these numbers (and labels)

* What does "notifications processed" mean? is this on the OZ side or the Notifications API side?
* The retry count relates to the request count in some way, maybe we should be able to describe what the lower and upper bound are? though we have a sliding window, a request that finished on second 59 can only be retried once with the 1s wait time before retry

"Configuration should be adjust in the admin..." -> I think we should clarify (and perhaps do a second test run to confirm) that you can do this be either increasing the delay factor or the maximum number of delays or both of those. It would be nice to get a working example of a configuration that can cover 5 minutes of down time:

E.g. without touching the factor/upper limit to delay duration:

1. At 0s request arrives and notification is scheduled
2. At 1s the first retry happens
3. At 2s the second
4. At 4s the third
5. At 8s the fourth
6. At 16s the fifth
7. At 32s the sixth
8. At 64s (capped to 48s) the sevent
9. At 96s the eight
10. At 144s the ninth
11. At 192s the tenth
12. At 240s the eleventh
13. At 288s the twelfth
14. At 332 the thirteenth

So that would imply a maximum retry count of 13 to bridge 5 minutes of downtime.